WordPress/gutenberg
# Rename refs to fix tons of 'Mutating a value' errors in react-compiler

## What?
This PR renames all refs we mutate to follow the ref naming pattern recommended in https://github.com/facebook/react/pull/29916.

## Why?
As part of #61788 we're working to experiment with `react-compiler` and early resolve any errors that can be blockers for using it. 

Renaming the refs to follow the naming pattern recommended in https://github.com/facebook/react/pull/29916 takes care of 53 of the 140 lint errors.

I'm landing that as a separate improvement from #61788 to keep it minimal, but also because ref rename is pretty harmless on its own.

## How?
We're renaming refs to follow the `somethingRef` pattern:

* `const something = useRef()` becomes `const somethingRef = useRef()`
* `const exampleRef = useRef()` remains unchanged

## Testing Instructions
* Verify Gutenberg still builds and all checks are green.
* There should be no functional changes, this is just a minor code style change.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 
None
